src/nimony/sigmatch.nim

```diff
@@ -807,6 +828,30 @@ proc isEmptyLiteral*(n: Cursor): bool =
     skip n # type
     result = n.kind == ParRi
 
+proc isEmptyCall*(n: Cursor): bool =
+  if n.exprKind notin CallKinds:
+    return false
+  var n = n
+  inc n
+  # overload of `@` with empty array param:
+  result = n.kind == Symbol and pool.syms[n.symId] == "@.1." & SystemModuleSuffix
```

Comment:
What if the `@` was overloaded? Still correct here then as it's done after overload resolution?

Reply:
Yes, this requires the argument to be semchecked already in a context that allows empty containers (can clarify this in a comment), which should disambiguate to the overload that takes an empty array. It would only break if `@` is overriden with an overload that matches an empty array better than the system one. If we remove the empty overload we could also check for the first overload of `@`, we could also just check for identifier equality to `@`.

Reply:
No, it's fine this way then.